This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tony Theodore <tonyt@logyst.com>
Date: Sat, 18 Nov 2017 20:50:55 +1100
Subject: [PATCH 1/1] Fix broken Makefile, add relevant object files and don't
 use -DCRYPTOPP_EXPORTS


diff --git a/GNUmakefile b/GNUmakefile
index 1111111..2222222 100755
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -255,8 +255,7 @@ TESTOBJS := bench.o bench2.o test.o validat1.o validat2.o validat3.o adhoc.o dat
 LIBOBJS := $(filter-out $(TESTOBJS),$(OBJS))
 
 # List cryptlib.cpp first in an attempt to tame C++ static initialization problems
-DLLSRCS := cryptlib.cpp algebra.cpp algparam.cpp asn.cpp basecode.cpp cbcmac.cpp channels.cpp des.cpp dessp.cpp dh.cpp dll.cpp dsa.cpp ec2n.cpp eccrypto.cpp ecp.cpp eprecomp.cpp files.cpp filters.cpp fips140.cpp fipstest.cpp gf2n.cpp gfpcrypt.cpp hex.cpp hmac.cpp integer.cpp iterhash.cpp misc.cpp modes.cpp modexppc.cpp mqueue.cpp nbtheory.cpp oaep.cpp osrng.cpp pch.cpp pkcspad.cpp pubkey.cpp queue.cpp randpool.cpp rdtables.cpp rijndael.cpp rng.cpp rsa.cpp sha.cpp simple.cpp skipjack.cpp strciphr.cpp trdlocal.cpp
-DLLOBJS := $(DLLSRCS:.cpp=.export.o)
+DLLOBJS := cryptlib.o cpu.o hex.o serpent.o randpool.o sharkbox.o blowfish.o files.o seal.o gzip.o fips140.o eprecomp.o dsa.o dll.o md2.o fipstest.o rc5.o mars.o osrng.o queue.o xtrcrypt.o gf256.o vmac.o authenc.o default.o idea.o squaretb.o pkcspad.o emsa2.o sha.o network.o eax.o tea.o square.o dessp.o crc.o elgamal.o zlib.o ida.o dh2.o camellia.o strciphr.o ecp.o marss.o adler32.o wait.o iterhash.o safer.o base32.o rc2.o polynomi.o oaep.o gf2n.o zinflate.o shark.o xtr.o shacal2.o cmac.o eccrypto.o rdrand.o ripemd.o gcm.o pubkey.o basecode.o trdlocal.o salsa.o esign.o luc.o hrtimer.o ttmac.o cbcmac.o mqv.o dh.o md4.o whrlpool.o rw.o cast.o asn.o tiger.o rng.o pssr.o misc.o mqueue.o algparam.o base64.o zdeflate.o channels.o ccm.o des.o blumshub.o ec2n.o casts.o gfpcrypt.o gost.o algebra.o rijndael.o modes.o sha3.o panama.o nbtheory.o md5.o socketft.o twofish.o tigertab.o tftables.o rsa.o rdtables.o sosemanuk.o 3way.o gf2_32.o rabin.o bfinit.o filters.o integer.o seed.o rc6.o wake.o skipjack.o arc4.o hmac.o winpipes.o 
 
 # Import lib testing
 LIBIMPORTOBJS := $(LIBOBJS:.o=.import.o)
